src/DIPS.Xamarin.UI/Extensions/ObjectExtensions.cs

```diff
@@ -13,15 +13,34 @@ public static class ObjectExtensions
         /// <returns></returns>
         public static string GetPropertyValue(this object obj, string propertyName)
         {
+            if (obj == null)
+            {
+                return string.Empty;
+            }
+
             if (string.IsNullOrEmpty(propertyName))
             {
-                return obj == null ? string.Empty : obj.ToString();
+                return obj.ToString();
             }
 
             var displayMember = obj.GetType().GetProperty(propertyName);
 
             var value = displayMember?.GetValue(obj, null);
             return value == null ? string.Empty : value.ToString();
         }
+
+        /// <summary>
+        /// Extracts a double value if one, else returns the tostring as a double. If none of these return a valid value, the default value is used
```

Comment:
"Tries to extract a double value from a property on an object. If the property is not a double value, it will use the ToString(). If the ToString is not a double value it will use the defaultValue parameter."